src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadPropertiesStepExecution.java

```diff
@@ -70,7 +72,11 @@ protected ReadPropertiesStepExecution(@Nonnull ReadPropertiesStep step, @Nonnull
             FilePath f = ws.child(step.getFile());
             if (f.exists() && !f.isDirectory()) {
                 try(InputStream is = f.read()){
-                   properties.load(is);
+                    if(StringUtils.isEmpty(step.getEncoding())){
+                        properties.load(is);
+                    } else {
+                        properties.load( new BufferedReader(new InputStreamReader(is, step.getEncoding())));
```

Comment:
You need to close the streams or it will break things, especially on windows.
Something like above:

```suggestion
                        try(InputStreamReader is = new InputStreamReader(is, step.getEncoding()); BufferedReader br = new BufferedReader(is)) {
                            properties.load(br);
                        }    
```